optlang/interface.py

```diff
@@ -76,6 +76,56 @@
 }
 
 
+def evaluate_expression(expression):
+    return expression.subs([
+        (sym, sym.value) for sym in expression.free_symbols])
+
+
+def handle_symbols(expression, instance, attr):
+    if isinstance(expression, sympy.Expr):
```

Comment:
The reference to sympy will not work for symengine. Maybe just use a try to check if it quacks like an expression (and maybe a check with is_numeric first)